test/PerformanceTests/SerializationComparisonsTests/JsonWriterBenchmarks/Benchmarks.cs

```diff
@@ -40,26 +39,24 @@ public Benchmarks()
             model = DataModel.GetEdmModel();
         }
 
-        [IterationSetup]
-        public void SetupStreams()
+        [GlobalSetup]
+        public void Setup()
         {
-            memoryStream = new MemoryStream();
-            string path = Path.GetTempFileName();
-            fileStream = new FileStream(path, FileMode.Create, FileAccess.Write);
             writer = writerCollection.GetWriter(WriterName);
+            filePath = Path.GetTempFileName();
         }
 
-        [IterationCleanup]
-        public void CleanUp()
+        [GlobalCleanup]
+        public void Cleanup()
         {
-            fileStream.Close();
-            fileStream = null;
+            File.Delete(filePath);
         }
 
         [Benchmark]
         [BenchmarkCategory("InMemory")]
         public async Task WriteToMemoryAsync()
         {
+            using var memoryStream = new MemoryStream();

```

Comment:
Shouldn't this happen in `IterationSetup` method?

Reply:
Yeah, it could also happen in `IterationSetup`. I thought doing it here woud help clean up the code. Also the cost of creating and disposing a memory stream should be relatively small since it doesn't involve any actual I/O. But the reason I moved it out of the `IterationSetup` is because the docs warn against using the `IterationSetup` for "microbenchmarks": https://benchmarkdotnet.org/articles/features/setup-and-cleanup.html